libs/labelbox/src/labelbox/schema/user_group.py

```diff
@@ -79,338 +127,584 @@ def __init__(
         id: str = "",
         name: str = "",
         color: UserGroupColor = UserGroupColor.BLUE,
-        users: Set[User] = set(),
-        projects: Set[Project] = set(),
-    ):
-        """
-        Initializes a UserGroup object.
+        description: str = "",
+        notify_members: bool = False,
+        default_role: Optional[Role] = None,
+        users: Optional[Set[User]] = None,
+        members: Optional[Set[UserGroupMember]] = None,
+        projects: Optional[Set[Project]] = None,
+    ) -> None:
+        """Initialize a UserGroup instance.
 
         Args:
-            client (Client): The Labelbox client object.
-            id (str, optional): The ID of the user group. Defaults to an empty string.
-            name (str, optional): The name of the user group. Defaults to an empty string.
-            color (UserGroupColor, optional): The color of the user group. Defaults to UserGroupColor.BLUE.
-            users (Set[User], optional): The set of users in the user group. Defaults to an empty set.
-            projects (Set[Project], optional): The set of projects associated with the user group. Defaults to an empty set.
+            client: Labelbox client for API communication.
+            id: Unique identifier (empty for new groups).
+            name: Display name for the group.
+            color: Visual color identifier.
+            description: Optional description.
+            notify_members: Whether to notify members of changes.
+            default_role: Default role for users added via legacy users field.
+            users: Legacy set of users for backward compatibility.
+            members: Set of members with explicit roles.
+            projects: Set of associated projects.
         """
         super().__init__(
             client=client,
             id=id,
             name=name,
             color=color,
-            users=users,
-            projects=projects,
+            description=description,
+            notify_members=notify_members,
+            default_role=default_role,
+            users=users or set(),
+            members=members or set(),
+            projects=projects or set(),
         )
 
-    def get(self) -> "UserGroup":
+    def model_post_init(self, __context: Any) -> None:
+        """Set default_role to LABELER if not specified.
+
+        Args:
+            __context: Pydantic context (unused).
         """
-        Reloads the user group information from the server.
+        if self.default_role is None:
+            try:
+                roles = self.client.get_roles()
+                self.default_role = roles.get("LABELER")
```

Comment:
I don't have a bigger picture here , but can we not make the `default_role` required ?